### ISSUE TITLE
TIF: read after save fails with EOFException

### ISSUE THREAD

**Issue by THausherr** [haraldk/TwelveMonkeys#241]
I am unable to read the file from #237 after saving it as a TIF:
```
        BufferedImage bim = ImageIO.read(new File("XXXX\\badccittstream.tif"));
        ImageIO.write(bim, "tif", new File("XXX\\badccittstream-saved.tif"));
        ImageIO.read(new File("XXXX\\badccittstream-saved.tif"));
```
[badccittstream.zip](https://github.com/haraldk/TwelveMonkeys/files/254628/badccittstream.zip)

```
Exception in thread "main" java.io.EOFException
	at javax.imageio.stream.ImageInputStreamImpl.readFully(ImageInputStreamImpl.java:353)
	at javax.imageio.stream.ImageInputStreamImpl.readFully(ImageInputStreamImpl.java:361)
	at com.twelvemonkeys.imageio.metadata.exif.EXIFReader.readValue(EXIFReader.java:328)
	at com.twelvemonkeys.image io.metadata.exif.EXIFReader.readValueAt(EXIFReader.java:287)
	at com.twelvemonkeys.imageio.metadata.exif.EXIFReader.readEntry(EXIFReader.java:277)
	at com.twelvemonkeys.imageio.metadata.exif.EXIFReader.readDirectory(EXIFReader.java:109)
	at com.twelvemonkeys.imageio.metadata.exif.EXIFReader.readDirectory(EXIFReader.java:133)
	at com.twelvemonkeys.imageio.metadata.exif.EXIFReader.read(EXIFReader.java:87)
	at com.twelvemonkeys.imageio.plugins.tiff.TIFFImageReader.readMetadata(TIFFImageReader.java:171)
	at com.twelvemonkeys.imageio.plugins.tiff.TIFFImageReader.readIFD(TIFFImageReader.java:252)
	at com.twelvemonkeys.imageio.plugins.tiff.TIFFImageReader.read(TIFFImageReader.java:715)
	at javax.imageio.ImageIO.read(ImageIO.java:1448)
	at javax.imageio.ImageIO.read(ImageIO.java:1308)
```

The saved file cannot be displayed with IrfanView too.

**Comment by THausherr** [comment]
I just see that there was a remark in #237 "I suppose the Encoder is also affected" (I didn't get that mail).

**Comment by Schmidor** [comment]
I've looked into that and was convinced that the encoder didn't need the additional array length.
I have to look into that, but as ImageIO.write doesn't conserve meta data it could be that the writer is writing uncompressed and you're running into #229 
This is what I encountered at writing #239

**Comment by Schmidor** [comment]
Indeed the writer defaults to uncompressed, if no compression mode is set by metadata or ImageWriterParams. So this is a duplicate of #229

**Comment by haraldk** [comment]
I was just about to comment on that, but you beat me to it.

Yes, I believe so too. 

Actually, it could be that we should use different defaults for compression depending on input image (b/w -> CCITT, uint RGB/Gray/CMYK -> LZW/PackBits, etc), but I think the current implementation will be fastest and most compatible (we should probably do the same thing as the JAI TIFFImageWriter does), at the expense of larger file sizes. But that is a different issue.


-- 
Harald K

**Comment by haraldk** [comment]
I believe this is fixed in 04a3915.

This will be part of the 3.3 release. Working hard on squashing the last known bugs....


Best regards,

-- 
Harald K

**Comment by haraldk** [comment]
@THausherr  This bugfix is now included in the official 3.3 release.
